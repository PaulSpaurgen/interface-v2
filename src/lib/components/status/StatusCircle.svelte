<script lang="ts">
	import type { CommonVariant } from '$lib/types/componentTypes';
	import { getColorHexByVariant } from '$lib/utils/helpers/componentHelper';

	export let variant: CommonVariant = 'primary';
	export let styleClass = '';

	const color = getColorHexByVariant(variant);
	$: style = `${styleClass} w-5 h-5 rounded-full mx-auto`;
</script>

<div class={style} style={`background-color:${color}`} />
