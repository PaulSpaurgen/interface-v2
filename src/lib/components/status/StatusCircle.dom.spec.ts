import { cleanup, render } from '@testing-library/svelte';
import StatusCircle from './StatusCircle.svelte';

describe('Status Circle', () => {
	afterEach(() => {
		cleanup();
		vi.resetAllMocks();
	});

	it('renders with default props', () => {
		const { getByTestId, container } = render(StatusCircle);
		expect(getByTestId('status-circle-test')).toBeTruthy();
		expect(container).toMatchSnapshot();
	});

	it('renders with custom props', () => {
		const { getByTestId } = render(StatusCircle, { props: { styleClass: 'status-circle-class' } });
		expect(getByTestId('status-circle-test')).toBeTruthy();
		expect(getByTestId('status-circle-test').classList.contains('status-circle-class')).toBe(true);
	});
});
