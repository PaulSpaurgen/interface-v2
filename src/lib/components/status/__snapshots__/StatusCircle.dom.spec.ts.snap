// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Status Circle > StatusCircle component renders with default porps 1`] = `
<body>
  <div>
    <div
      class="mx-auto h-5 w-5 rounded-full "
      data-testid="status-circle-test"
    />
    <!--&lt;StatusCircle&gt;-->
  </div>
</body>
`;
